### ISSUE TITLE
Dependencies between nested engines

### ISSUE THREAD

**Issue by catz** [ember-engines/ember-engines#527]
Hello,

Can you please help me to describe dependencies for following structure?

```
Host Application ->
   service:ajax
   Nested 1 (async routable) ->
     service:socket
     Nested 1.1 (async routable)
       -> how to pass socket service via dependencies from 1 to 1.1 ?
     Nested 1.2 
   Nested 2 (async)
```

In host app I could write something like this in app.js, but the trouble is that socket service is not available until I load it.

```
Nested1.1: {
    dependencies: {
      services: ['ajax', 'socket']
    }
  }
```
   
I tried to write it in Nested1 engine.js file, but it seems Nested1.1 is looking for dependencies in Host app and not in parent engine ? service:socket is quite big and I wouldn't want to move it on Host level.

Thank you.

**Comment by catz** [comment]
Small demo app: https://github.com/catz/eng-test

User object from wall-engine is shared to nested host-engine via app service property.

**Comment by lvegerano** [comment]
I'm running into this issue as well but with `externalRoutes` dependencies. I have an engine that I mount to the host app and also to another engine that is mounted to the same host app.

```
Host App
    mounts: Engine A
    mounts: Engine B
        mounts: Engine A
```
```
Engine A deps:
   externalRoutes: {
        'route-dep': 'host.app.route',
    }
```
``` 
Engine B deps
   Engine A deps
       externalRoutes: {
           'route-dep': 'engine.b.route'
       }
```

But when `Engine A` tries to route to `route-dep` from `Engine B`, instead of routing to `engine.b.route` it routes to `host.app.route` Is there a way around this?

**Comment by swarnavallis** [comment]
Is there a way to share services from parent engine to child engine (nested engines)

**Comment by catz** [comment]
@swarnavallis 
> Is there a way to share services from parent engine to child engine (nested engines)

Did you check http://ember-engines.com/guide/services Or see my initial question

**Comment by swarnavallis** [comment]
Yeah my use case is .. I have a common host app under which i have say 4 engines which are each a separate logical application.Now i have splitted each engine to n engines(nested engine). I have a service specific to one logical application which is the parent engine. Now this service will be needed in the child engine, why should i move a service specific to one engine to host app? Its better if child engine is able to use the parent engines' services if not what is the working behaviour of nested engine. @catz

**Comment by catz** [comment]
@swarnavallis  and this is exactly what I asked but didn't find any practical solution without sharing via host app.

**Comment by swarnavallis** [comment]
> @swarnavallis and this is exactly what I asked but didn't find any practical solution without sharing via host app.

So is there any possibility that this will be addressed in near future? @catz  @stefanpenner
